lib/services/role.go

```diff
@@ -1401,7 +1400,7 @@ func isFormatOld(cert *ssh.Certificate) bool {
 	_, hasRoles := cert.Extensions[teleport.CertExtensionTeleportRoles]
 	_, hasTraits := cert.Extensions[teleport.CertExtensionTeleportTraits]
 
-	if hasRoles && hasTraits {
+	if hasRoles || hasTraits {
```

Comment:
Just making sure you intended to change the logic here, ignore if it's on purpose.

Reply:
on purpose